controllers/controllers/clusters/vsphere.go

```diff
@@ -135,5 +137,49 @@ func (v *VSphereClusterReconciler) Reconcile(ctx context.Context, cluster *anywh
 		return reconciler.Result{}, err
 	}
 
+	workerNodeGroupMachineSpecs := make(map[string]anywherev1.VSphereMachineConfigSpec, len(machineConfigMap))
+	for _, wnConfig := range cluster.Spec.WorkerNodeGroupConfigurations {
+		workerNodeGroupMachineSpecs[wnConfig.MachineGroupRef.Name] = machineConfigMap[wnConfig.MachineGroupRef.Name].Spec
+	}
+
+	cp := machineConfigMap[specWithBundles.Spec.ControlPlaneConfiguration.MachineGroupRef.Name]
+	var etcd *anywherev1.VSphereMachineConfigSpec
+	if specWithBundles.Spec.ExternalEtcdConfiguration != nil {
+		etcdSpec := machineConfigMap[specWithBundles.Spec.ExternalEtcdConfiguration.MachineGroupRef.Name]
```

Comment:
is the naming up side down?

```suggestion
etcd := machineConfigMap[specWithBundles.Spec.ExternalEtcdConfiguration.MachineGroupRef.Name]
etcdSpec = &etcd.Spec
```

Reply:
done